src/RobotsTxt.php

```diff
@@ -126,36 +126,49 @@ protected function getDisallowsPerUserAgent(string $content): array
 
         $disallowsPerUserAgent = [];
 
-        $currentUserAgent = null;
+        $currentUserAgents = [];
+
+        $treatAllowDisallowLine = false;
 
         foreach ($lines as $line) {
-            if ($this->isCommentLine($line)) {
+            if ($this->isCommentOrEmptyLine($line)) {
                 continue;
             }
 
             if ($this->isUserAgentLine($line)) {
+                if ($treatAllowDisallowLine) {
+                    $treatAllowDisallowLine = false;
+                    $currentUserAgents = [];
+                }
                 $disallowsPerUserAgent[$this->parseUserAgent($line)] = [];
 
-                $currentUserAgent = &$disallowsPerUserAgent[$this->parseUserAgent($line)];
+                $currentUserAgents[] = &$disallowsPerUserAgent[$this->parseUserAgent($line)];
 
                 continue;
             }
 
-            if ($currentUserAgent === null) {
+            if ($this->isDisallowLine($line)) {
+                $treatAllowDisallowLine = true;
+            } elseif ($this->isAllowLine($line)) {
+                $treatAllowDisallowLine = true;
+                continue;
+            } else {
                 continue;
             }
 
             $disallowUrl = $this->parseDisallow($line);
 
-            $currentUserAgent[$disallowUrl] = $disallowUrl;
+            foreach ($currentUserAgents as &$currentUserAgent) {
+                $currentUserAgent[$disallowUrl] = $disallowUrl;
+            }
         }
 
         return $disallowsPerUserAgent;
     }
 
-    protected function isCommentLine(string $line): bool
+    protected function isCommentOrEmptyLine(string $line): bool
     {
-        return strpos(trim($line), '#') === 0;
+        return strpos(trim($line), '#') === 0 || trim($line) === '';
```

Comment:
I don't like complex if statements. Could you you split the if and use early returns?

Reply:
split done